OWML.ModHelper.Input/InputInterceptor.cs

```diff
@@ -0,0 +1,84 @@
+﻿using UnityEngine;
+
+namespace OWML.ModHelper.Input
+{
+    internal class InputInterceptor
+    {
+        public static void SingleAxisRemovePre(SingleAxisCommand __instance)
+        {
+            ModInputHandler.Instance.UnregisterGamesBinding(__instance);
+        }
+
+        public static void DoubleAxisRemovePre(DoubleAxisCommand __instance)
+        {
+            ModInputHandler.Instance.UnregisterGamesBinding(__instance);
+        }
+
+        public static void SingleAxisUpdatePost(
+            SingleAxisCommand __instance,
+            ref float ____value,
+            int ____axisDirection,
+            KeyCode ____gamepadKeyCodePositive,
+            KeyCode ____gamepadKeyCodeNegative,
+            KeyCode ____keyPositive,
+            KeyCode ____keyNegative
+        )
+        {
+            KeyCode positiveKey, negativeKey;
+            ModInputHandler.Instance.RegisterGamesBinding(__instance);
+            int axisDirection = 1;
+            if (OWInput.UsingGamepad())
+            {
+                axisDirection = ____axisDirection;
+                positiveKey = ____gamepadKeyCodePositive;
+                negativeKey = ____gamepadKeyCodeNegative;
+            }
+            else
+            {
+                positiveKey = ____keyPositive;
+                negativeKey = ____keyNegative;
+            }
+            if (UnityEngine.Input.GetKey(positiveKey) && ModInputHandler.Instance.ShouldIgnore(positiveKey))
```

Comment:
You check `UnityEngine.Input.GetKey(someKey) && ModInputHandler.Instance.ShouldIgnore(someKey)` a lot, could it be a private method?

Reply:
integrated